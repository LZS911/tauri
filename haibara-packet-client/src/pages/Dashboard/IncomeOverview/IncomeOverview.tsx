import { Column, ColumnConfig } from '@ant-design/plots';
import { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import useRenderCustomPlots from '../../../customHooks/useRenderCustomPlots';
import { getWeekLocales } from '../Common';

const mockLastWeekData: () => ColumnConfig['data'] = () => {
  return getWeekLocales().map((w) => {
    return {
      date: w,
      value: Math.floor(Math.random() * 200),
    };
  });
};

const config: ColumnConfig = {
  data: [],
  xField: 'date',
  yField: 'value',
};

const IncomeOverview: React.FC = () => {
  const { t } = useTranslation();
  const customProps = useRenderCustomPlots();
  const [data, setData] = useState<ColumnConfig['data']>([
    { date: '', value: '' },
  ]);
  useEffect(() => {
    setData(mockLastWeekData());
  }, []);
  return (
    <div className="mb-5 max-w-[100%] basis-[100%] lg:max-w-[40%] lg:basis-[40%]">
      <div className="mb-4 flex items-center justify-between leading-[1.625rem] dark:text-white">
        <h2 className="font-semibold dark:text-white">
          {t('dashboard.incomeOverview.title')}
        </h2>

        <span className="text-xs">
          {t('dashboard.incomeOverview.subTitle')}
        </span>
      </div>

      <Column {...{ ...customProps, ...config, data }} />
    </div>
  );
};

export default IncomeOverview;
