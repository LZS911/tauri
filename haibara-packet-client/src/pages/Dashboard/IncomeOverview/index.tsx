import IncomeOverview from './IncomeOverview';

export default IncomeOverview;
