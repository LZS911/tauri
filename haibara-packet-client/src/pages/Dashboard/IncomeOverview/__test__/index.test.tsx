import { renderWithRedux } from '../../../../testUtils/customRender';
import IncomeOverview from '../index';

describe('test IncomeOverview', () => {
  test('should match snapshot', () => {
    const { container } = renderWithRedux(<IncomeOverview />);
    expect(container).toMatchSnapshot();
  });
});
