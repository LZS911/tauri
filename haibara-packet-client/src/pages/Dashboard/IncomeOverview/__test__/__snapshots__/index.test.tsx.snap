// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test IncomeOverview should match snapshot 1`] = `
<div>
  <div
    class="mb-5 max-w-[100%] basis-[100%] lg:max-w-[40%] lg:basis-[40%]"
  >
    <div
      class="mb-4 flex items-center justify-between leading-[1.625rem] dark:text-white"
    >
      <h2
        class="font-semibold dark:text-white"
      >
        dashboard.incomeOverview.title
      </h2>
      <span
        class="text-xs"
      >
        dashboard.incomeOverview.subTitle
      </span>
    </div>
    <div
      class="rounded-md border border-stone-200 p-5 dark:border-stone-700 dark:bg-darkMode bg-white"
      data-chart-source-type="G2Plot"
      size-sensor-id="1"
      style="height: inherit;"
    >
      <div
        style="position:relative;"
      >
        <canvas
          height="400"
          style="width: 400px; height: 400px; display: inline-block; vertical-align: middle;"
          width="400"
        />
      </div>
      <object
        class="size-sensor-object"
        data="about:blank"
        style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
        tabindex="-1"
        type="text/html"
      />
    </div>
  </div>
</div>
`;
